subsys/bluetooth/services/ddfs.c

```diff
@@ -277,23 +280,24 @@ static void ctrl_point_ind(struct bt_conn *conn, uint8_t req_op, uint8_t status,
 		net_buf_simple_add_mem(&buf, data, len);
 	}
 
-	ddfs_inst.ind_params.attr = &ddf_svc.attrs[DDF_SVC_CTRL_POINT_ATTR_IDX];
+	ddfs_inst.ind_params.attr = attr;
 	ddfs_inst.ind_params.func = indicate_cb;
 	ddfs_inst.ind_params.destroy = indicate_destroy;
 	ddfs_inst.ind_params.data = ind;
 	ddfs_inst.ind_params.len = buf.len;
 
-	if (bt_gatt_indicate(conn, &ddfs_inst.ind_params) == 0) {
+	err = bt_gatt_indicate(conn, &ddfs_inst.ind_params);
+	if (err) {
+		LOG_ERR("Failed to send Control Point indication, err: %d", err);
+	} else {
 		ddfs_inst.indicating = true;
 	}
 }
 
 int bt_ddfs_distance_measurement_notify(struct bt_conn *conn,
 					const struct bt_ddfs_distance_measurement *measurement)
 {
-	if (!ddfs_inst.dm_notification_enabled) {
-		return -EACCES;
-	}
```

Comment:
Why don't we want to check if notification is enabled?

Reply:
I do check just before sending notification by calling `bt_gatt_is_subscribed`